Dockerfile

```diff
@@ -1,9 +1,11 @@
 FROM node:alpine
 
-COPY . /app
 WORKDIR /app
 
-RUN apk add git && npm install
+COPY package*.json ./
 
-CMD npm start
+RUN npm ci --only=production
```

Comment:
This allows Docker to cache `node_modules/` when `package.json` and `package-lock.json` don't change, correct?

Reply:
This is correct... Each command (RUN, COPY and ADD) creates a new layer for your docker image cache.
As soon as any file in the current command changes, the cache is invalid, and it will runn the step without cache.